src/msw/toolbar.cpp

```diff
@@ -1503,19 +1508,17 @@ bool wxToolBar::MSWCommand(WXUINT WXUNUSED(cmd), WXWORD id_)
     // global variable: if it gets reset from our dtor, we will know that the
     // toolbar was destroyed by this handler and that we can't use this object
     // any more.
-    gs_liveToolbar = this;
+    gs_liveToolbars.insert(this);
 
     bool allowLeftClick = OnLeftClick(id, toggled);
 
-    if ( gs_liveToolbar != this )
+    if ( gs_liveToolbars.find(this) == gs_liveToolbars.end() )
     {
         // Bail out, we can't touch any member fields in the already
         // destroyed object anyhow.
         return true;
     }
 
-    gs_liveToolbar = NULL;
```

Comment:
Shouldn't we call `gs_liveToolbars.erase()` here? I don't think anything catastrophic will happen if we don't, but it seems logically wrong to let a toolbar whose events are not being processed in this hash set.

And it's not like it costs us anything, as we already call `find()` anyhow.